### ISSUE TITLE
Error with displaying results in sqlops - Query failed: Conversion overflows.

### ISSUE THREAD

**Issue by eilerth** [microsoft/azuredatastudio#275]
- SQL Operations Studio Version: 0.23.6

Steps to Reproduce:
Execute this query: SELECT CAST(80 AS DECIMAL(38, 27))
Rather than returning a result value of 80, I get the error message "Query failed: Conversion overflows."

I can dump that value into a temp table with no problem:
SELECT CAST(80 AS DECIMAL(38, 27)) [a] INTO #a

And then if I try to query the temp table I get the same issue.
If I then cast as FLOAT it returns value just fine:
SELECT CAST([a] AS FLOAT) FROM #a

**Comment by kevcunnane** [comment]
@saurabh500 this looks at first glance to be a .net core issue. This works fine in SSMS (regular .Net Stack) but fails for us. Stack trace is below - it's reading a decimal into an object buffer using `SqlDataReader.GetValues(object[] values);`.

Could you help verify this is a dotnet issue rather than something we control, and if so get it tracked in corefx? Stacktrace is below, this is pretty easy to repro.

```csharp
Exception thrown: 'System.OverflowException' in System.Data.SqlClient.dll: 'Conversion overflows.'
   at System.Data.SqlClient.SqlBuffer.get_Decimal()
   at System.Data.SqlClient.SqlBuffer.get_Value()
   at System.Data.SqlClient.SqlDataReader.GetValueFromSqlBufferInternal(SqlBuffer data, _SqlMetaData metaData)
   at System.Data.SqlClient.SqlDataReader.GetValues(Object[] values)
   at Microsoft.SqlTools.ServiceLayer.QueryExecution.DataStorage.StorageDataReader.GetValues(Object[] values) in /Users/me/projects/sqltoolsservice/src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/StorageDataReader.cs:line 116
```

**Comment by saurabh500** [comment]
@kevcunnane I see the same behavior where an exception is thrown on .Net Framework as well. The precision of upto 26 is supported both on .Net core and framework. 

```C#
using System;
using System.Data.Common;
using System.Data.SqlClient;

namespace SqlClientTest
{
    class Program
    {
        static void Main(string[] args)
        {
            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder
            {
                DataSource = "localhost",
                IntegratedSecurity = true
            };
            using (SqlConnection connection = new SqlConnection(builder.ToString()))
            {
                connection.Open();
                DbCommand cmd = connection.CreateCommand();
                cmd.CommandText = "SELECT CAST(80 AS DECIMAL(38, 27))"; // 38,26 works
                DbDataReader reader = cmd.ExecuteReader();
                while (reader.Read())
                {
                    Object[] values = new Object[reader.FieldCount];

                    int result = reader.GetValues(values);
                    Console.WriteLine(values[0]);
                }
            }

        }
    }
}
```

**Comment by kburtram** [comment]
It looks like SSMS calls `DbDataReader::GetSqlValues` instead of `DbDataReader::GetValues`.  `GetValues` throws an `System.Overflow` exception whereas `GetSqlValues` correctly returns the result.  Unfortunately, simply changing our reader to call `GetSqlValues` breaks a bunch of other stuff.

Given the impact of issue relative to the size of the change this is something we'll have to pick-up later when there's time to refactor the data reader.

**Comment by eilerth** [comment]
Thanks a bunch for looking into this and sending the update!

I may just be showing the level of my naivety, but for a little hack until
a better fix happens could we just wrap the GetValues in a try/catch and if
it catches this specific error try to get the value with GetSqlValues and
then round it to a supported length (if needed) and kick back into whatever
data type GetValues was returning (if different)?

Certainly a small issue, but if a simple little hack like that "fixes" it
for now, that'd be cool.

Thanks again!

**Comment by ranasaria** [comment]
+1 on eilerth’s suggestion.


-  Arvind.

________________________________

**Comment by philosophicles** [comment]
FWIW this issue on the vscode-mssql project is probably the same thing: https://github.com/Microsoft/vscode-mssql/issues/1190

Cross-linking for visibility when people like me search, and in case it helps to bring these different application teams together!

**Comment by philosophicles** [comment]
Slight advance on the impact of this bug. 

This trivial query works as expected:
```
with cte as (
    select cast(-114.75000 as numeric(38,5)) as RawNum
)
select  RawNum
        --, 1/RawNum
from cte
```

However, running with the `1/RawNum` line uncommented:
```
with cte as (
    select cast(-114.75000 as numeric(38,5)) as RawNum
)
select  RawNum
        , 1/RawNum
from cte
```

returns

| RawNum | (No column name) |
| --- | --- |
| NULL | NULL |

With the `Query failed: Conversion overflows.` error returned as a message. 

So the failure in the second column is also affecting the results in the first (and any/all other) columns of the result set. This seems like a possibly-preventable problem to me? Aside from @eilerth 's suggestion, can anything be done to prevent an error with one column affecting rendering of all the others?

**Comment by eilerth** [comment]
In my opinion I'd prefer not to have some of the values still displayed while the overflows are not, I'd rather it be obvious there is a problem with the rendered results so I don't mistakenly use the data that isn't complete.

But I can see value in that suggestion, so thanks for commenting, @philosophicles . Also thanks for linking to the vscode issue for visibility.

**Comment by philosophicles** [comment]
> I'd rather it be obvious there is a problem with the rendered results so I don't mistakenly use the data 

I can also understand that point of view :)

**Comment by alanrenmsft** [comment]
the fix for the issue introduced an issue with money column type and needs to be reverted to unblock the Dec release. reopen the issue and will try to fix it in the next release.

**Comment by alanrenmsft** [comment]
@leandroko could you please open a new issue with detailed descriptions?

**Comment by leandroko** [comment]
@alanrenmsft sure! thanks!
